Gordon360/Services/NewsService.cs

```diff
@@ -93,13 +110,14 @@ public StudentNews SubmitNews(StudentNews newsItem, string username, string id)
             var categoryIDParam = new SqlParameter("@CategoryID", newsItem.categoryID);
             var subjectParam = new SqlParameter("@Subject", newsItem.Subject);
             var bodyParam = new SqlParameter("@Body", newsItem.Body);
+            var imageParam = new SqlParameter("@Image", "");
 
             // Run stored procedure
-            var result = RawSqlQuery<StudentNewsViewModel>.query("INSERT_NEWS_ITEM @Username, @CategoryID, @Subject, @Body", usernameParam, categoryIDParam, subjectParam, bodyParam);
-            if (result == null)
-            {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The data was not found." };
-            }
+            IEnumerable<int> idResult = _context.Database.SqlQuery<int>("INSERT_NEWS_ITEM @Username, @CategoryID, @Subject, @Body, @Image", usernameParam, categoryIDParam, subjectParam, bodyParam, imageParam);
+
+            int snid = idResult.FirstOrDefault();
```

Comment:
This is my fault for misinforming you, but since we expect exactly one value in this `IEnumerable`, we should really be using `idResult.Single();`. That will throw an exception if there is more than one `SNID` in idResult, which is exactly what we want.